extensions/avro/src/main/java/com/hazelcast/jet/avro/impl/AvroSchemaStorage.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.avro.impl;
+
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.EntryAdapter;
+import com.hazelcast.core.EntryEvent;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.LifecycleEvent;
+import com.hazelcast.map.IMap;
+import org.apache.avro.Schema;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.hazelcast.jet.impl.JobRepository.INTERNAL_JET_OBJECTS_PREFIX;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.HOURS;
+import static org.apache.avro.SchemaNormalization.parsingFingerprint64;
+
+/**
+ * Schemas are stored in an IMap, where the key is the 64-bit Rabin fingerprint of the
+ * canonical JSON representation of the schema. As explained in the
+ * <a href="https://avro.apache.org/docs/1.11.1/specification/#schema-fingerprints">Avro
+ * specification</a>, such fingerprints are safe to be used as a key in schema caches of up
+ * to a million entries (for such a cache, the chance of a collision is 3x10<sup>-8</sup>).
+ * <p>
+ * Schemas that are not used for one week are dropped and the expiration time of the most
```

Comment:
>Schemas that are not used for one week are dropped

There is no good TTL for schemas. Records may be kept for long time in window aggregation processors (eg. if the window is 2 weeks).

Reply:
also a serialized record can be kept in buffer (Outbox or Inbox) for a long time - usually not desired but theoretically possible. If the schema expires between `put` and `get`, entire job will fail